mgmt/utils/MgmtSocket.cc

```diff
@@ -154,6 +155,25 @@ mgmt_open_mode(const char *path, int oflag, mode_t mode)
 }
 
 //-------------------------------------------------------------------------
+// mgmt_open_mode_elevate
+//-------------------------------------------------------------------------
+
+int
+mgmt_open_mode_elevate(const char *path, int oflag, mode_t mode, bool elevate_p)
+{
+  int r, retries;
+  for (retries = 0; retries < MGMT_MAX_TRANSIENT_ERRORS; retries++) {
+    r = elevate_p ? elevating_open(path, oflag, mode) : ::open(path, oflag, mode);
+    if (r >= 0) {
+      return r;
+    }
+    if (!mgmt_transient_error()) {
+      break;
+    }
+  }
+  return r;
+}
```

Comment:
Socket...why...just, why?

Meanwhile, back at the ranch, is this retry loop maybe involved in the hammering of Tomcat that @d2r observed?

Reply:
No.

Reply:
I think this particular instance is not related. Created #2591 to look into maybe doing a back-off in cases similar to the one in which I observed the problem.